### ISSUE TITLE
Improvement based on Signposting the Scholarly Web (http://www.signposting.org/)

### ISSUE THREAD

**Issue by huberrob** [pangaea-data-publisher/fuji#65]
1. By now we only test typed links in the HEAD section of the landing page. We should also test for links included in the header response following the signposting.org convention (e.g., get/head request)
2. There will be a proposal from @hvdsomp on the final recommendations on implementing link relation types (including linkset) in the data landing page. To test the recommendations, we need (a) pilot repositories (e.g., pangaea) implementing them and fuji assessment should be extended to reflect the final recommendations.

**Comment by uschindler** [comment]
We were one of the first repositories having signposting. Is there some changes, I an do them, but in general links are sent since approx 2016.

**Comment by kitchenprinzessin3880** [comment]
@uschindler , yes.
@hvdsomp is working some changes to the original proposal, so once the final recommendation is available from him, we can update the implementation.

**Comment by uschindler** [comment]
The typed links in HTML head are identical to the HTTP headers. So if one changes the other ones, too (it's just a different formatting of exact same signposting information).

**Comment by hvdsomp** [comment]
Currently working on a FAIR Signposting Profile. If the Patterns on signposting.org are considered a spec, then the Profile will be an implementation guideline. It builds on typed links in `HTTP Link` and `HTML head`, but also on Link Sets as specified in [this Internet Draft](https://datatracker.ietf.org/doc/draft-wilde-linkset/) (soon to be an RFC).

**Comment by ajaunsen** [comment]
@uschindler mentions early implementations of signposting, which I presume is PANGEA. Can you provide an example for the sake of illustration?

**Comment by huberrob** [comment]
Yes this is implemented in PANGAEA basically for every dataset. A GET or HEAD request for https://doi.pangaea.de/10.1594/PANGAEA.884028

will return the following header which is  including the signposting links:

nginx/1.19.5
date:
Wed, 09 Dec 2020 07:17:02 GMT
content-type:
text/html;charset=utf-8
content-length:
86468
connection:
keep-alive
vary:
Accept; Cookie, X-PanLoginID, Authorization
cache-control:
public
link:
<https://doi.org/10.1594/PANGAEA.884028>;rel="cite-as", <https://doi.pangaea.de/10.1594/PANGAEA.884028?format=metadata_jsonld>;rel="describedby";type="application/ld+json", <https://doi.pangaea.de/10.1594/PANGAEA.884028?format=citation_ris>;rel="describedby";type="application/x-research-info-systems", <https://doi.pangaea.de/10.1594/PANGAEA.884028?format=citation_bibtex>;rel="describedby";type="application/x-bibtex", <https://doi.pangaea.de/10.1594/PANGAEA.884028?format=html>;rel="item";type="text/html", <https://doi.pangaea.de/10.1594/PANGAEA.884028?format=textfile>;rel="item";type="text/tab-separated-values", <https://orcid.org/0000-0003-3601-9072>;rel="author", <https://orcid.org/0000-0002-8683-1854>;rel="author"
x-powered-by:
Jetty(9.4.33.v20201020)
strict-transport-security:
max-age=31536000
x-ua-compatible:
IE=Edge
x-content-type-options:
nosniff
x-frame-options:
SAMEORIGIN

**Comment by huberrob** [comment]
Just a reminder to not forget the rel=item thing which also can be used to indicate data objects

**Comment by hvdsomp** [comment]
hi all, the [FAIR Signposting spec](https://signposting.org/FAIR/) is meanwhile in really good shape. It would be good to have a look, if possible. Many aspects have remained unchanged when compared to the early days of Signposting when @uschindler was one of the first adopters. But there are some new aspects too:

- Additional typed links, i.e. `type` and `license`
- A "logic" that prescribes to only provide certain types of links (`cite-as`, `describedby`, `license`, `type`, `author`) for content resources when the information is different than what is provided for the scholarly object as a whole by typed links at the landing page
- Ability to provide links pertaining to all resources that make up a scholarly object in a standalone document called a Link Set, which is made discoverable by a link with a `linkset` relation type ([standardization of Link Set document formats - including a JSON format - and relation type at IETF near completion](https://datatracker.ietf.org/doc/draft-ietf-httpapi-linkset/)).

**Comment by huberrob** [comment]
Dear Herbert,

I recently was pointed to OAI-ORE you also have initiated some years ago (#149) and there seem to be at least some overlaps at the 'item' level. Can you please explain the relations between the link set doc approach and OAI-ORE ... if any. 

Robert

**Comment by hvdsomp** [comment]
Dear Robert,

That's a pertinent question, which I will try to answer as concise as possible:

First, OAI-ORE and (FAIR) Signposting definitely share a goal, namely to express in a machine-actionable manner aspects of a scholarly object as it exists on the web including listing its constituent resources (aggregated resources in OAI-ORE, content resources or items in FAIR Signposting). There are important differences too:

- OAI-ORE (completed 2008) is focused on describing a scholarly object (think metadata-like). It leverages semantic web and linked data approaches and as such allows highly expressive descriptions using terms from the OAI-ORE vocabulary as well as from others. Obviously, implementations require a software stack that can handle such semantic descriptions. 
- Signposting (started 2017) is focused on navigating a scholarly object (hence the Signposting name). It leverages REST/HATEOAS/follow-your-nose principles (merely using typed links with a limited set of IANA-registered link relation types) to allow bots to discover resources that are part of or related to the scholarly object, and to navigate to them to obtain further information. Because the very basic technological approach taken by Signposting, I sometimes jokingly refer to it as the poor man's OAI-ORE.

Link Sets ([spec here](https://datatracker.ietf.org/doc/draft-ietf-httpapi-linkset/)) are a general purpose approach to convey a collection of typed links pertaining to a resource in a stand-alone document. Two document formats, each with their dedicated MIME type are specified. Link Sets fit squarely in the REST/HATEOAS/follow-your-nose approach (they are themselves discoverable using a link with `linkset` relation type) and become important, for example, when the number of typed links to be expressed becomes too large to handle in an HTTP header or to convey links pertaining to resources for which one does not have access to the HTTP header and hence can't add HTTP Links there. Signposting is just one of the possible use cases for Link Sets but, again, this is a general purpose technology. For example, recently GS1 (massive barcode publisher) released a specification in which Link Sets play a significant role and will become used in global services related to barcodes, see [this announcement](https://groups.google.com/g/gs1-digital-link-developers/c/2JiqcmAM164).
